btclib/psbt.py

```diff
@@ -109,10 +181,10 @@ def decode(cls: Type[_PsbtIn], input_map: Dict[bytes, bytes]) -> _PsbtIn:
             elif key[0] == 0x06:
                 assert len(key) == 33 + 1
                 assert len(value) % 4 == 0
-                hd_keypaths[key[1:].hex()] = {
-                    "fingerprint": value[:4].hex(),
-                    "derivation_path": decode_der_path(value[4:]),
-                }
+                path = value[4:]
+                hd_keypaths.add_hd_path(
+                    key[1:], value[:4], unpack("<" + "I" * int((len(path) / 4)), path)
```

Comment:
The problem here is that i have to create a list of bytes from a raw bytes array. With unpack i can create a string expression in one line. Using int.from_bytes i have to split bytes in groups. I should rewrote the line

unpack("<" + "I" * int((len(path) / 4)), path)

with something like:

res = []
for i in range(0, len(path), 4):
    res.append(int.from_bytes(path[i : i+4]))

I'm not sure that this could be easier to read. What do you think?

Reply:
In my opinion int.from_bytes is more readable, because it is very clear what the function does without looking at the documentation (as an example I hadn't understood that the output of unpack in this case was a list of integers). On the other hand int.from_bytes does make the code more verbose. So we have to choose between readability and verbosity

Reply:
I would avoid unpack:
res = [int.from_bytes(path[i : i+4] for i in range(0, len(path), 4)]

Reply:
Done. But sourcery say that with this addition the code quality is decreased by 0.02% :D . In my opinion the code quality calculation is not very reliable :)